numba/core/types/abstract.py

```diff
@@ -170,22 +191,34 @@ def augment(self, other):
     # (returns True) or `types.int32(types.int32[:])` (returns something
     # usable as a function signature).
 
-    def __call__(self, *args):
+    @pt.overload
+    def __call__(self, *args: NumbaTypeInst) -> "Signature": ...
+
+    @pt.overload
+    def __call__(self, *args: pt.Any) -> pt.Any: ...
+
+    def __call__(self, *args: pt.Any) -> pt.Any:
         from numba.core.typing import signature
         if len(args) == 1 and not isinstance(args[0], Type):
             return self.cast_python_value(args[0])
         return signature(self, # return_type
                          *args)
 
-    def __getitem__(self, args):
+    def __getitem__(self, args: pt.Union[slice,
```

Comment:
Is this the type of formatting we'd like to use? I'm personally a fan of black style formatting.